wave_matching/include/wave/matching/matcher.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef __WAVE_MATCHING_MATCHER_HPP__
+#define __WAVE_MATCHING_MATCHER_HPP__
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+namespace wave {
+namespace matching {
+
+// sensor data type
+template <typename R>
```

Comment:
Would have used `T`instead of `R` for typename, unless you have a good reason to choose `R`?